src/extension/gutters.ts

```diff
@@ -42,17 +40,9 @@ export class Gutters {
             );
             if (!pickedReport) { throw new Error("Could not show Coverage Report file!"); }
 
-            // Construct the webview panel for the coverage report to live in
-            const previewPanel = window.createWebviewPanel(
-                "coverageReportPreview",
-                "Preview Coverage Report",
-                ViewColumn.One,
-            );
+            const previewPanel = new PreviewPanel(pickedReport);
 
-            // Read in the report html and send it to the webview
-            const reportUri = Uri.file(pickedReport);
-            const reportHtml = await workspace.openTextDocument(reportUri);
-            previewPanel.webview.html = reportHtml.getText();
+            await previewPanel.createWebView();
```

Comment:
### Discussion
We might have to store the preview panel on the instance so we can can call dispose later (`public dispose()`).
I have a bit of pseudo code below, let me know what you think.

### Research
https://code.visualstudio.com/api/references/vscode-api#WebviewPanel

### Pseudo Code
```typescript
this.previewPanel = new PreviewPanel(pickedReport);

~~~~

public dispose() {
    this.previewPanel.dispose();
}
```

Reply:
> Looking great! Just a couple minor comments and a couple discussion points.
> 
> ### Regarding Testing:
> I believe you also have the testing working that we have for this (just integration ones really) but we could also add a simple unit style one to make sure the `addContentSecurityPolicy` properly appends the policy. We could also consider then what happens when it can't find the meta tag (does it gracefully fail?)

@ryanluker Yes, the integration test is working. I will work on writing the unit tests.

Reply:
@ryanluker this is how I want to go about writing the unit test to check if `addContentSecurityPolicy` properly appends the policy. 

1. Create a variable that holds an HTML text.
2. Make the `addContentSecurityPolicy` property of PreviewPanel public, so that I can have access to it in the test.
3. Check if the returned value from calling `addContentSecurityPolicy` contains the content security policy header.

What do you think about my approach? I also think it is more logical to check for the presence of a _head_ tag than a _meta_ tag?

Reply:
That approach sounds great @Kaytbode ! you could also add a failing unit test to confirm what we expect to happen in that scenario 🤔 .

In terms of approach, I am not sure which is more common in these preview reports (we only have the node generated one so there could be some variability). We could also support both options (check for meta, if not found try head but still add just one security policy).

Reply:
@ryanluker You are yet to review the changes I made

Reply:
@Kaytbode Thanks for the reminder, I will review again shortly.